mpitests/test_apply.py

```diff
@@ -556,7 +556,7 @@ def circuit():
 
         cpu_qnode = qml.QNode(circuit, dev_cpu)
         expected_output_cpu = cpu_qnode()
-        comm.Bcast(expected_output_cpu, root=0)
+        comm.Bcast(np.array(expected_output_cpu), root=0)
```

Comment:
is this casting necessary?

hmm.. will it be not working without casting to a numpy array? :thinking: 
 
```suggestion
        comm.Bcast(expected_output_cpu, root=0)
```

Reply:
Not without fixing something else.

Reply:
what is this "something else"? :)

Reply:
I assume this is due to MPI4PY having special buffers for numpy data. I suspect the output is Pythonic lists/tuples, which don't use this buffer, so the cast is needed

Reply:
Thanks @mlxd! I'll continue this in pv to avoid blocking merge of this PR.